data/crac/crac-api/src/test/java/com/powsybl/openrao/data/cracapi/NetworkActionUtils.java

```diff
@@ -372,19 +236,25 @@ public static NetworkElement createNetworkElement(String networkElementId) {
         return new NetworkElementImplTest(networkElementId);
     }
 
-    public static TopologicalAction createTopologyAction(NetworkElement networkElement, ActionType actionType) {
-        return new TopologicalActionImplTest(networkElement, actionType);
+    public static NetworkAction createSwitchAction(NetworkElement networkElement, ActionType actionType) {
+        return new NetworkActionImplTest(new SwitchActionBuilder().withId("id").withNetworkElementId(networkElement.getId()).withOpen(actionType == ActionType.OPEN).build());
     }
 
-    public static SwitchPair createSwitchPair(NetworkElement switchToOpen, NetworkElement switchToClose) {
-        return new SwitchPairImplTest(switchToOpen, switchToClose);
+    public static NetworkAction createSwitchPair(NetworkElement switchToOpen, NetworkElement switchToClose) {
+        return new NetworkActionImplTest(new SwitchPairImplTest(switchToOpen, switchToClose));
     }
 
-    public static PstSetpoint createPstSetpoint(NetworkElement pst, int setpoint) {
-        return new PstSetpointImplTest(pst, setpoint);
+    public static NetworkAction createPhaseTapChangerTapPositionAction(NetworkElement pst, int setpoint) {
+        return new NetworkActionImplTest(new PhaseTapChangerTapPositionActionBuilder().withId("id").withNetworkElementId(pst.getId()).withTapPosition(setpoint).withRelativeValue(false).build());
     }
 
-    public static InjectionSetpoint createInjectionSetpoint(NetworkElement networkElement, double setpoint, Unit unit) {
-        return new InjectionSetpointImplTest(networkElement, setpoint, unit);
+    public static NetworkAction createGeneratorAction(NetworkElement networkElement, double setpoint, Unit unit) {
```

Comment:
Not very important because that's only used in test but base the selection of action on a Uni is not very explicit.
 It may be a good idea to split into two function, createGeneratorTargetVAction and createGeneratorActivePowerAction

Reply:
done